api/src/main/java/org/openmrs/api/db/TagDAO.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.api.db;
+
+import org.openmrs.OpenmrsObject;
+import org.openmrs.Tag;
+import org.openmrs.api.APIException;
+
+import java.util.List;
+
+public interface TagDAO {
+	
```

Comment:
You will need to update methods in this DAO based on the changes in the service in regard to the comments